### PR TITLE
Updated default relevant logic for multi select case lists

### PR DESCRIPTION
**Pull request by orangejenny**
## Technical Summary
Branched off of https://github.com/dimagi/commcare-hq/pull/31341, only the two last commits are new

Multi-select case lists don't use the default relevant condition. Relevancy will be handled by the `exclude` expression in the `<data>` for the case id.

https://dimagi-dev.atlassian.net/browse/USH-1817

## Feature Flag
USH: Allow selecting multiple cases from the case list

### Safety story
Small addition that isn't going to be used in prod yet. Automated tests should be sufficient for regression testing.

### Automated test coverage

Tests in PR. Suite tests cover regressions.

### QA Plan

None for this PR. This feature will be QAed later.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/app_manager/suite_xml/post_process/remote_requests.py [hunk 1]
```diff
@@ -352,7 +352,8 @@ def __init__(self, suite, module, detail_section_elements, endpoint_id, case_ses
         self.case_session_var = case_session_var
 
     def get_post_relevant(self):
-        return CaseClaimXpath(self.case_session_var).default_relevant()
+        if not self.module.case_details.short.multi_select:
+            return CaseClaimXpath(self.case_session_var).default_relevant()
```
Comment (snopoke) [inline corehq/apps/app_manager/suite_xml/post_process/remote_requests.py:356]: why doesn't this use `module.get_relevant`?

File: corehq/apps/app_manager/tests/test_suite_remote_request.py [hunk 2]
```diff
@@ -187,14 +187,22 @@ def setUp(self):
...
 
         self.assertEqual(config.get_relevant(), "count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/search_case_id]) = 0")  # noqa: E501
 
         config.additional_relevant = "double(now()) mod 2 = 0"
         self.assertEqual(config.get_relevant(), "(count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/search_case_id]) = 0) and (double(now()) mod 2 = 0)")  # noqa: E501
 
+    def test_search_config_relevant_multi_select(self, *args):
+        config = CaseSearch()
+
+        self.assertEqual(config.get_relevant(multi_select=True), None)
```
Comment (snopoke) [inline corehq/apps/app_manager/tests/test_suite_remote_request.py:201]: Is the `None` return value correctly handled by the callers?

File: corehq/apps/app_manager/suite_xml/post_process/remote_requests.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/app_manager/suite_xml/post_process/remote_requests.py:356]: `get_relevant` isn't on `Module`, it's on `CaseSearch`, and this code doesn't have a case search config.

Although I could move `get_relevant` up to module and name it something like `get_claim_relevant`, which would reduce the `module.case_details.short.multi_select` checks. I'll try that out.

File: corehq/apps/app_manager/tests/test_suite_remote_request.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/app_manager/tests/test_suite_remote_request.py:201]: Good question, and yes. [This call](https://github.com/dimagi/commcare-hq/blob/669d6ef243dfd69aff3eac870e92f99bb503dd31/corehq/apps/app_manager/suite_xml/post_process/remote_requests.py#L105-L107) checks for None, and the other calls of it ultimately get passed to `get_all_instances_referenced_in_xpaths` which filters out Nones.

File: corehq/apps/app_manager/suite_xml/post_process/remote_requests.py (hunk 1 above)
Comment (snopoke) [reply corehq/apps/app_manager/suite_xml/post_process/remote_requests.py:356]: no need to change it since there is a good reason

**Review by snopoke — APPROVED** [review]

**Pull request merged**